mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoUpdateMethodGenerator.java

```diff
@@ -0,0 +1,189 @@
+import java.util.Optional;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+
+public class DaoUpdateMethodGenerator extends DaoMethodGenerator {
+
+  private static final EnumSet<ReturnTypeKind> SUPPORTED_RETURN_TYPES =
+      EnumSet.of(VOID, FUTURE_OF_VOID, RESULT_SET, FUTURE_OF_ASYNC_RESULT_SET);
+
+  public DaoUpdateMethodGenerator(
+      ExecutableElement methodElement,
+      DaoImplementationSharedCode enclosingClass,
+      ProcessorContext context) {
+    super(methodElement, enclosingClass, context);
+  }
+
+  @Override
+  public Optional<MethodSpec> generate() {
+
+    // Validate the parameters:
+    // - the first one must be the entity.
+    // - the others are completely free-form (they'll be used as additional bind variables)
+    if (methodElement.getParameters().isEmpty()) {
+      context
+          .getMessager()
+          .error(
+              methodElement,
+              "Wrong number of parameters: %s methods must have at least one",
+              Update.class.getSimpleName());
+      return Optional.empty();
+    }
+    VariableElement firstParameter = methodElement.getParameters().get(0);
+    TypeElement entityElement = asEntityElement(firstParameter);
+    if (entityElement == null) {
+      context
+          .getMessager()
+          .error(
+              methodElement,
+              "Invalid parameter type: "
+                  + "%s methods must take the entity to update as the first parameter",
+              Update.class.getSimpleName());
+      return Optional.empty();
+    }
+
+    // Validate the return type:
+    ReturnType returnType = parseReturnType(methodElement.getReturnType());
+    if (!SUPPORTED_RETURN_TYPES.contains(returnType.kind)) {
+      context
+          .getMessager()
+          .error(
+              methodElement,
+              "Invalid return type: %s methods must return either void or the entity class "
+                  + "(possibly wrapped in a CompletionStage/CompletableFuture)",
+              Update.class.getSimpleName());
+      return Optional.empty();
+    }
+    if (returnType.entityElement != null && !returnType.entityElement.equals(entityElement)) {
+      context
+          .getMessager()
+          .error(
+              methodElement,
+              "Invalid return type: %s methods must return the same entity as their argument ",
+              Update.class.getSimpleName());
+      return Optional.empty();
+    }
+
+    // Generate the method:
+    String helperFieldName = enclosingClass.addEntityHelperField(ClassName.get(entityElement));
+    String statementName =
+        enclosingClass.addPreparedStatement(
+            methodElement,
+            (methodBuilder, requestName) ->
+                generatePrepareRequest(methodBuilder, requestName, helperFieldName));
+
+    MethodSpec.Builder insertBuilder = GeneratedCodePatterns.override(methodElement);
```

Comment:
Nit: this should be called `updateBuilder` for consistency. Although I've been wondering if it really needs to be named after the type of method, `methodBuilder` would be good too.